### ISSUE TITLE
validate cog login before push

### ISSUE THREAD

**Issue by anotherjesse** [replicate/cog#1221]
as part of #1157 - we could validate (using replicate apis, using r8.im directly?) that you have a valid cog authentication before pushing to r8.im

this is particularly painful when you have a system where you have to use `sudo cog push` as it is super easy to have accidentally just do `cog login` instead of `sudo cog login`

the error you get is confusing:

```
name unknown: The model https://replicate.com/xxx/yyy does not exist
```

**Comment by anotherjesse** [comment]
cc: @daanelson @joehoover

**Comment by teaguexiao** [comment]
This is also my case, thanks for saving me my day.

**Comment by mattt** [comment]
This should now be fixed by https://github.com/replicate/cog/pull/1733, which is available in [v0.9.10](https://github.com/replicate/cog/releases/tag/v0.9.10)
